lib/xgboost.rb

```diff
@@ -59,32 +63,36 @@ def train(params, dtrain, num_boost_round: 10, evals: nil, early_stopping_rounds
       end
 
       num_boost_round.times do |iteration|
+        cb_container.before_iteration(model: booster, epoch: iteration)
         booster.update(dtrain, iteration)
 
-        if evals.any?
```

Comment:
Please keep the existing code where possible to keep the changeset minimal / easier to review.

Reply:
Sorry, disabled rubocop